### PR TITLE
Add Loan to Model component

### PR DESCRIPTION
**Pull request by seeziyang**

### CODE REVIEW

File: src/main/java/seedu/address/storage/loanrecords/JsonAdaptedLoan.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import seedu.address.model.borrower.BorrowerId;
+import seedu.address.model.loan.Loan;
+
+/**
+ * Jackson-friendly version of {@link Loan}.
+ */
+class JsonAdaptedLoan {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Loan's %s field is missing!";
+    public static final String DATE_MESSAGE_CONSTRAINTS = "Date should be in the ISO-8601 format uuuu-MM-dd";
```
Comment (hoholyin) [inline src/main/java/seedu/address/storage/loanrecords/JsonAdaptedLoan.java]: Should we change "uuuu" to "yyyy" for better readability (and understandability)?

**Review by hoholyin — COMMENTED** [review]
Looks good! 👍

File: src/main/java/seedu/address/MainApp.java [hunk 2]
```diff
@@ -34,8 +34,8 @@
 import seedu.address.storage.borrowerrecords.JsonBorrowerRecordsStorage;
 import seedu.address.storage.catalog.CatalogStorage;
 import seedu.address.storage.catalog.JsonCatalogStorage;
-import seedu.address.storage.loanrecord.JsonLoanRecordsStorage;
```
Comment (shiyao821) [inline src/main/java/seedu/address/MainApp.java:37]: Good on spotting the spelling mistake!

**Comment by seeziyang** [comment]
Closes #49

**Pull request merged**